### PR TITLE
Add native graphql source RFC

### PR DESCRIPTION
**Pull request by freiksenet**
[view formatted RFC](https://github.com/gatsbyjs/rfcs/blob/native-graphql-support/text/0000-native-graphql-source.md)

### CODE REVIEW

**Comment by m-allanson** [comment]
[View formatted](https://github.com/gatsbyjs/rfcs/blob/native-graphql-support/text/0000-native-graphql-source.md).

@freiksenet added an example project (linked from the RFC). If you want to jump straight to how this might look in your project:

- [Setup in `gatsby-config.js`](https://github.com/freiksenet/gatsby-github-displayer/blob/master/gatsby-config.js)
- [A page query](https://github.com/freiksenet/gatsby-github-displayer/blob/master/src/pages/index.js)

**Comment by freiksenet** [comment]
@sedubois nothing changed in this respect, it's a normal gatsby data source. So it's meant to be used to statically generate stuff.

**Comment by LeKoArts** [comment]
How would this work with "external" packages, like:
https://github.com/braposo/figma-graphql

Would this also be possible? In this case you wouldn't need to create a source plugin.

For the rest this source plugin would replace the need for a source plugin when a GrpahQL endpoint is existent, right?

**Comment by freiksenet** [comment]
I believe that it replace many of the existing source plugins that only work with one graphql api.

**Comment by LeKoArts** [comment]
And about the first part of my question?

**Comment by freiksenet** [comment]
@LeKoArts Ah, sorry missed that part of the question. This will work with any GraphQL endpoint, so you wouldn't need a source plugin.

File: text/0000-native-graphql-source.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+# Detailed design
+
+## Action Creator
+
+`addThirdPartySchema(schema: GraphQLSchema)`
+
+Adds a schema to third-party schema list. At schema creation time all the schemas are merged with [schema stitching](https://www.apollographql.com/docs/graphql-tools/schema-stitching.html)
+
+## graphql-source-graphql
```
Comment (zcei) [inline text/0000-native-graphql-source.md]: Shouldn't this be `gatsby-source-graphql` as in the examples above, or is this another plugin?

**Comment by swim** [comment]
Yay this is pretty fantastic, super keen to see the introduced methods (addThirdPartySchema). Is it only fragments which have the typename prepended? 

```
... on [typeName]_FragementName
... on Wordpress_WordPressAcf_quote {
```

**Comment by freiksenet** [comment]
Also argument declarations. All fields names are unchanged. I had to err on
side of caution there, otherwise even trivial schema conflicted.

**Comment by wyze** [comment]
@KyleAMathews had reached out to me and see if we could work together on this @freiksenet, but it looks like you did all the work already.

I'm willing to donate the package name I have on `npm`, but not sure the steps involved in that. I can always republish it under my named scope on npm if I want to continue using mine instead of the official one.

**Comment by jlengstorf** [comment]
Hey @wyze! The steps to change package ownership are pretty painless, fortunately. 😄 

```
npm owner add kylemathews gatsby-source-graphql
```

Once @KyleAMathews is an owner, we can handle the rest.

Thanks so much for donating the package name!

**Comment by freiksenet** [comment]
@wyze ping?

**Comment by wyze** [comment]
Should be good to go.

**Comment by jlengstorf** [comment]
Thanks so much, @wyze! Can I send you some swag as a thank you? You can email me at jason@gatsbyjs.com and/or let me know your email and I'll get you sorted.

Thanks again!

**Comment by wyze** [comment]
Who is going to deny swag!?! I'll shoot you an email. Thanks Jason.

**Comment by magicspon** [comment]
BELTING!

**Comment by sandys** [comment]
this is very needed. Quick question - will the source-filesystem plugin usecase also be served through this ? I would prefer it otherwise filesystem queries would drift away from how everything else gets queried. 

also - this RFC is too important for it to be part of a "plugin". if this will be part of core, then do we really need it as a  `resolve: "gatsby-source-graphql", ` ?

**Comment by magicspon** [comment]
Should we be using this?
https://www.npmjs.com/package/gatsby-source-graphql

**Comment by wyze** [comment]
@magicspon That still points to my package. I think you need to wait for https://github.com/gatsbyjs/gatsby/pull/6475 to get merged and published.

**Comment by KyleAMathews** [comment]
The PR for this RFC has been merged so moving this RFC PR into its final comment period. Speak now or forever hold your peace! If all still looks good, we'll merge this in 3 days.

**Comment by abumalick** [comment]
This is amazing progress for Gatsby :rocket:

**Comment by magicspon** [comment]
i've been using `gatsby-source-graphql` with craftcms... works like a charm!

**Comment by KyleAMathews** [comment]
And we're in! Thanks @freiksenet and others!

**Pull request merged**